### PR TITLE
Fix issue #285 - Unexpected targets' cleanup order

### PR DESCRIPTION
**Pull request by facundofc**

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -21,6 +21,7 @@ Changes
  - Add configuration `DOIT_CONFIG` `action_string_formatting` to control command action formatter.
  - Fix `result_dep`, use result **after** its execution
  - Fix #286: Cannot use functools.partial as title
+ - Fix #285: Unexpected targets' cleanup order
```
Comment (schettino72) [inline CHANGES]: CHANGES entries should read like "what was fixed", not like "issue description".
Same could be said about git commit

- Fix #286: Cannot use functools.partial as title
+ Fix #286: Support functools.partial on task's dict `metadata task.title`
- Fix #285: Unexpected targets' cleanup order
+ Fix #285: clean command, remove files before attempting to remove folders

File: tests/test_task.py [hunk 2]
```diff
@@ -433,6 +437,19 @@ def test_keep_non_empty_dirs(self, tmpdir):
             assert expected == os.path.exists(filename)
         assert os.path.exists(tmpdir['dir'])
 
+    def test_clean_in_reverse_lexical_order(self, tmpdir):
```
Comment (schettino72) [inline tests/test_task.py]: The test name is misleading because it actually does not ensure reverse lexical order.

**Comment by facundofc** [comment]
> thanks.
> 
> sorry if review sounds too nitpick.
> I can do these changes later myself when merging.

Hey! I don't find it nitpick, I think it's understandable to wish to keep some consistency.

File: CHANGES (hunk 1 above)
Comment (facundofc) [reply CHANGES]: Understood. I had the impression that the other changes quoted the issue's title, but I didn't actually go and check.

I changed the phrasing of #285 solution, let me know what you think.

File: tests/test_task.py (hunk 2 above)
Comment (facundofc) [reply tests/test_task.py]: I agree. I'm having a bit of a tough time finding a proper name though. An option I considered is to actually do ensure that the order is that one, by patching both `os.rmdir` and `os.remove`, what would you think about this?

File: CHANGES [hunk 3]
```diff
@@ -20,7 +20,8 @@ Changes
  - Make it possible to use a custom encoder when using config_changed with a dict.
  - Add configuration `DOIT_CONFIG` `action_string_formatting` to control command action formatter.
  - Fix `result_dep`, use result **after** its execution
- - Fix #286: Cannot use functools.partial as title
+ - Fix #286: Support `functools.partial` on tasks' dict metadata `task.title`
+ - Fix #285: `clean` command, remove targets in reverse DFS order
```
Comment (schettino72) [inline CHANGES]: What is DFS? "Distributed File System"?

**Comment by schettino72** [comment]
merged. thanks

File: CHANGES (hunk 3 above)
Comment (facundofc) [reply CHANGES]: It's Depth First Search in this case. I didn't mean to be pedantic.

Thanks for merging!

**Pull request merged**